public/pages/DetectorsList/List/List.tsx

```diff
@@ -138,8 +146,16 @@ export const DetectorList = (props: ListProps) => {
       search: queryString.stringify(updatedParams),
     });
 
-    dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
+    const getUpdatedDetectors = async () => {
+      try {
+        await dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
+      } catch (error) {
+        console.log('Error is found during getting detector list', error);
+        setIsLoadingFinalDetectors(false);
```

Comment:
should we move`setIsLoadingFinalDetectors(false);` to `finally`? looks like only when exception is caught, `isLoading` can be set to false

Reply:
That's the behavior I want here. It should only be set to false if there was an error. If successful and the detectors are retrieved, it should be set to true until the final set of sorted detectors to display are finished [here](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/public/pages/DetectorsList/List/List.tsx#L169).

Note that `isLoadingFinalDetectors` is only related to sorting/filtering in the frontend. `isRequestingFromES` handles loading from the backend.

Reply:
Got it.